README.md

```diff
@@ -2,173 +2,295 @@
+each scenario, so you won't be able to `recall` anything from previous scenarios.
 
-If you can not extend `ActorWorld`, you can also add an `ActorLookup` instance to your world object like so:
+### Accessing the world from actors
 
-```typescript
-import { ActorLookUp } from '@cucumber/screenplay'
+If your interactions need to access data in the `world`, they can do so via the `Actor#world` property. If you're doing this
+you should also declare the generic type of the actor in the interaction implementation:
 
-class World {
-  private readonly actorLookUp = new ActorLookUp()
-
-  findOrCreateActor(actorName: string): Actor<any> {
-    return this.actorLookUp.findOrCreateActor(this, actorName)
+```typescript
+export const moveTo: MoveTo = (coordinate) => {
+  // We're declaring the World type of the actor so that we can access its members
+  return async (actor: Actor<World>) => {
+    actor.world.shouty.moveTo(actor.name, coordinate)
   }
 }
 ```
 
-## Specifying different regular expressions for matching actors
+### Handling asynchronous behaviour
 
-The regular expression used to match an `Actor` name provided by `defineActorParameterType` only match a capital latin letter followed
-by any number of latin letters. This may not fit your needs, but it is pretty easy to write your own matcher.
+In a distributed system it may take some time before the outcome of an action propagates around the whole system.
 
-For example, if you want to only match some specific names for your actors:
+For example, in a chat application, when one user sends a message, it may take a few milliseconds before the 
+other users receive the message, because it travels through a network.
+
+In cases like this you can use the `eventually` function to periodically check for a specific condition:
 
 ```typescript
-import { setWorldConstructor, defineParameterType } from '@cucumber/cucumber'
-
-defineParameterType({
-  name: 'actor',
-  regexp: /Alice|Bob|Charlie/,
-  transformer(this: World, actorName: string) {
-    return this.findOrCreateActor(actorName)
-  },
+Then('{actor} hears {actor}’s message', async function (this: World, listener: Actor<World>, shouter: Actor) {
+  const shouterLastMessage = shouter.recall('lastMessage')
+
+  await eventually(() => {
+    const listenerMessages = listener.ask(this.inboxMessages())
+    assert.deepStrictEqual(listenerMessages, [shouterLastMessage])
+  })
 })
 ```
 
+The `eventually` function accepts a single argument - a zero argument `condition` function. If the `condition` function 
+throws an error, it will be called again at a regular `interval` until it passes without throwing an exception. If it doesn't
+pass or finish within a `timeout` period, a timeout error is thrown.
+
+The default `interval` is `50ms` and the default `timeout` is `1000ms`. This can be overridden with a second 
+`{ interval: number, timeout: number}` argument after the `condition`.
+
+### Design recommendations
```

Comment:
I don't think we should have the recommendations here, maybe more in a blog-post which covers good practices and a wide range of examples.
I think there might be too many cases possible for us to provide guidance. But I may be wrong.

Reply:
I've used the same principles in my (SmartBear-internal) pull request to the Example Mapping app - and they seem to hold up pretty well.

I've presented this stuff at several conferences, and the feedback I have received is that people lack guidance about how to do this, so I'd like to keep it for now. We can always revise it later.

Reply:
I do not doubt it works fine :)
But if you'd had enough feedback with people needing guidance, then this makes sense to have it here.

And if those explanations do not help some persons (because this system can not apply to their application), they can still seek some help somewhere else. At least it should be helpful in most cases.